extensions/amp-carousel/0.1/slidescroll.js

```diff
@@ -220,6 +220,22 @@ export class AmpSlideScroll extends BaseSlides {
     if (slide !== undefined) {
       this.showSlideWhenReady_(slide);
     }
+    const slideCount = mutations['slide-count'];
+    if (slideCount !== undefined && isFinite(slideCount)) {
+      let slideCountNum = Number(slideCount);
+      // Clamp count to between 0 and the total number of slides
+      slideCountNum = Math.max(Math.min(this.slides_.length, slideCountNum), 0);
```

Comment:
I don't think that's going to work. Under this PR `this.slides_.length` is the true number of slides in the carousel, whereas `this.noOfSlides_` becomes the number of slides the carousel is going to show. The change as proposed would fail if [slide-count] were set to lower than the true number of slides, and then raised back.